db/db_impl/db_impl.cc

```diff
@@ -3743,6 +3743,20 @@ void DBImpl::GetColumnFamilyMetaData(ColumnFamilyHandle* column_family,
   ReturnAndCleanupSuperVersion(cfd, sv);
 }
 
+void DBImpl::GetAllColumnFamilyMetaData(
+    std::vector<ColumnFamilyMetaData>* metadata) {
+  for (auto cfd : *(versions_->GetColumnFamilySet())) {
+    auto* sv = GetAndRefSuperVersion(cfd);
+    {
+      InstrumentedMutexLock l(&mutex_);
+      ColumnFamilyMetaData cf_meta;
+      metadata->emplace_back();
+      sv->current->GetColumnFamilyMetaData(&metadata->back());
+    }
+    ReturnAndCleanupSuperVersion(cfd, sv);
+  }
+}
+
```

Comment:
Since this is meant to be a generalization of `GetLiveFilesMetaData`, I think we should provide a consistent view of all column families the same way `GetLiveFilesMetaData` does (this could be achieved by locking the mutex outside the loop; this would also eliminate the need to reference each `SuperVersion`).

Reply:
I am not that familiar with the internals here.  What are you suggesting I use instead of GetAndRefSuperVersioni?

Reply:
You can simply access the current `Version` via `cfd->current()`. There's no need to increment/decrement the reference count assuming you do this under the DB mutex (since the `Version` cannot change while we're holding the lock).